lib/rc.js

```diff
@@ -2,11 +2,11 @@
 const fs = require('fs')
 
 function stringify (parsedData, commentChar) {
-  return Object.keys(parsedData).map(key => {
-    const v = parsedData[key]
-    const finalKey = v.comment ? `${commentChar}${key}` : key
-    return `${finalKey}=${v.value}`
-  }).join('\n') + '\n'
+  return parsedData.map(line => {
+    const finalKey = line.comment ? `${commentChar}${line.key}` : line.key
+    const separator = finalKey && line.value ? '=' : ''
```

Comment:
Every line gets turned into an array of objects, with a definite structure:

```
    if (parts.length === 2) {
      const { key, comment } = parseKey(parts[0].trim())
      const value = parts[1].trim()
      if (key) {
        result.push({ key, value, comment })
      }
    } else if (parts.length === 1) {
      const { key, comment } = parseKey(parts[0].trim())
      result.push({ key, value: '', comment })
    }
```

So this really isn't an issue. Also note that the test case at the end of `rc-test.js` covers this exact thing - it checks to see if the fixture (`tests/fixtures/pmuellr`) comes back intact after going through a serialize/deserialize cycle, and indeed it does.